doc/api/async_hooks.md

```diff
@@ -973,6 +973,10 @@ chains. It allows storing data throughout the lifetime of a web request
 or any other asynchronous duration. It is similar to thread-local storage
 in other languages.
 
+While you can create your own implementation on top of the `async_hooks` module,
+`AsyncLocalStorage` should be preferred as it is a performant and memory safe
+implementation that involves non-obvious optimizations.
```

Comment:
I would put more emphasis on expressing that there are _significant_ performance implications to implementing context storage in a way that is not carefully performance-tuned.

Reply:
👍 Pushed a new commit, it's now:

> `AsyncLocalStorage` should be preferred as it is a performant and memory safe implementation that involves significant optimizations that are non-obvious to implement.